apps/avatax/src/modules/saleor/order-parser.ts

```diff
@@ -14,3 +15,28 @@ export class SaleorOrderParser {
     return safeParse(payload);
   }
 }
+
+export class SaleorOrderCancelledParser {
```

Comment:
Lets call it "Factory". You can have a separate class or use factory method

```
class SaleorCancelledOrder {
private constructor() {}

static create() {}
}
```

factory method is better because you can make constructor private. This way, there is *no way to create class other way that via factory method*

Now, you give me "parser" but I can create instance of SaleorCancelledOrder that is not validated anyhow. So I see no value of this